alert-common/src/main/java/com/synopsys/integration/alert/common/channel/issuetracker/service/IssueTrackerRequestCreator.java

```diff
@@ -0,0 +1,157 @@
+            IssueTrackerRequest issueRequest = IssueResolutionRequest.of(issueSearchProperties, issueContentModel, alertIssueOrigin);
+            return List.of(issueRequest);
+        } else {
+            logger.debug("The top level action was not a DELETE action so it will be ignored");
+        }
+        return List.of();
+    }
+
+    protected List<IssueTrackerRequest> createRequestsByComponentGroup(ContentKey providerContentKey, String providerUrl, LinkableItem topic, LinkableItem nullableSubTopic,
+        SetMap<String, ComponentItem> groupedComponentItems) throws IntegrationException {
+        List<IssueTrackerRequest> issues = new LinkedList<>();
+
+        for (Set<ComponentItem> componentItems : groupedComponentItems.values()) {
+            ComponentItem arbitraryItem = componentItems
+                                              .stream()
+                                              .findAny()
+                                              .orElseThrow(
+                                                  () -> new AlertException(String.format("No actionable component items were found. Provider: %s, Topic: %s, SubTopic: %s", providerContentKey.getProviderName(), topic, nullableSubTopic)));
+            ItemOperation operation = arbitraryItem.getOperation();
+            String trackingKey = createAdditionalTrackingKey(arbitraryItem);
+            IssueSearchProperties issueSearchProperties = createIssueSearchProperties(providerContentKey.getProviderName(), providerUrl, topic, nullableSubTopic, arbitraryItem, trackingKey);
+
+            IssueTrackerRequest issueRequest = null;
+            AlertIssueOrigin alertIssueOrigin = new AlertIssueOrigin(providerContentKey, arbitraryItem);
+            if (ItemOperation.ADD == operation || ItemOperation.UPDATE == operation) {
+                IssueContentModel issueContentModel = issueTrackerMessageParser.createIssueContentModel(providerContentKey.getProviderName(), IssueCreationRequest.OPERATION, topic, nullableSubTopic, componentItems, arbitraryItem);
+                issueRequest = IssueCreationRequest.of(issueSearchProperties, issueContentModel, alertIssueOrigin);
+            } else if (ItemOperation.DELETE == operation) {
+                IssueContentModel issueContentModel = issueTrackerMessageParser.createIssueContentModel(providerContentKey.getProviderName(), IssueResolutionRequest.OPERATION, topic, nullableSubTopic, componentItems, arbitraryItem);
+                issueRequest = IssueResolutionRequest.of(issueSearchProperties, issueContentModel, alertIssueOrigin);
+            } else if (ItemOperation.INFO == operation) {
+                IssueContentModel issueContentModel = issueTrackerMessageParser.createIssueContentModel(providerContentKey.getProviderName(), IssueCommentRequest.OPERATION, topic, nullableSubTopic, componentItems, arbitraryItem);
+                issueRequest = IssueCommentRequest.of(issueSearchProperties, issueContentModel, alertIssueOrigin);
```

Comment:
Should this be changed to a switch case? Also the issueRequest line of each block seems identical. Shouldn't we create the isssueRequest after the conditonal blocks?

Reply:
Right now, I'm focusing on abstracting common code. Since this code existed and was functional, I am going to leave it for now and address it in a different PR.